packages/spearly-cms-js-core/src/Generator.ts

```diff
@@ -138,57 +138,74 @@ export class SpearlyJSGenerator {
             const patternName = result.attributes.patternName;
-            return [this.convertFromFieldsValueDefinitions(templateHtml, replacementArray, result, contentType, insertDebugInfo), uid, patternName]
+            let html = this.convertFromFieldsValueDefinitions(templateHtml, replacementArray, result, contentType, insertDebugInfo);
+            if (html.includes("cms-loop"))  {
+                html = await this.generateSubLoop(html, contentType, insertDebugInfo, result)
+            }
+            return [html, uid, patternName]
         } catch (e: any) {
             return Promise.reject(e);
         }
     }
 
-    async traverseInjectionSubLoop(nodes: HTMLElement[], apiOptions: APIOption, insertDebugInfo: boolean): Promise<Node[]> {
-        const resultNode = parse("") as HTMLElement
-        for (const node of nodes) {
-            const isTextNode = node.nodeType === 3
-            if (!isTextNode && node.hasAttribute("cms-loop")) {
-                const contentType = node.getAttribute("cms-field")
-                if (!contentType) {
-                    // Error case.
-                    return Promise.reject(new Error("cms-loop element doesn't have cms-field"))
-                }
-                const varName = node.getAttribute("cms-item-variable")
-                node.removeAttribute("cms-loop")
-                node.removeAttribute("cms-field")
-                node.removeAttribute("cms-item-variable")
-                const generatedStr = await this.generateList(node.outerHTML, contentType, varName || contentType, apiOptions, insertDebugInfo)
-                const generatedNode = parse(generatedStr) as HTMLElement
-                resultNode.childNodes = generatedNode.childNodes
-                continue
-            }
-            if (node.childNodes.length > 0) {
-                node.childNodes = await this.traverseInjectionSubLoop(node.childNodes as HTMLElement[], apiOptions, insertDebugInfo)
-            }
-            resultNode.appendChild(node)
+    getElementDepth(targetElement: HTMLElement, topElement: HTMLElement) {
+        let depth = 0
+        let currentElement = targetElement
+        while (currentElement !== topElement) {
+            depth++
+            currentElement = currentElement.parentNode as HTMLElement
         }
-        return resultNode.childNodes
+        return depth
     }
 
-    async generateSubLoop(templateHtml: string, apiOptions: APIOption, insertDebugInfo): Promise<string> {
-        const parsedNode = parse(templateHtml)
-        parsedNode.childNodes = await this.traverseInjectionSubLoop(parsedNode.childNodes as HTMLElement[], apiOptions, insertDebugInfo)
-        return parsedNode.outerHTML
+    async generateSubLoop(nodeHTML: string, contentTypeId: string, insertDebugInfo: boolean, parentData: Content): Promise<string> {
+        const node = parse(nodeHTML);
+        const cmsLoopNodes = node.querySelectorAll("[cms-loop]");
+
+        if (cmsLoopNodes.length > 0) {
```

Comment:
このままでももちろん問題ないのですが、これでもいいのかなと思ったので。

```suggestion
        if (cmsLoopNodes.length) {
```